Add a test that checks the included license files against included dependencies

wenottingham opened the pull request:
##### SUMMARY

We include licenses for our dependent python and JS code.
Ergo, add an automated tests so we don't miss when they need updated.

ryanpetrello commented on awx/main/tests/functional/test_licenses.py:
```diff
@@ -0,0 +1,135 @@
+    def read_api_requirements(path):
+        ret = {}
+        for file in ['requirements.txt', 'requirements_ansible.txt', 'requirements_git.txt', 'requirements_ansible_git.txt']:
+            fname = '%s/%s' % (path, file)
```
```python
from pip._internal.req import parse_requirements
for x in parse_requirements('./requirements.txt', session=''):
    print((x.name, str(x.specifier)))

('amqp', '==2.3.2')
('appdirs', '==1.4.2')
('argparse', '==1.4.0')
('asgi-amqp', '==1.1.2')
('asgiref', '==1.1.2')
('asn1crypto', '==0.24.0')
('attrs', '==17.4.0')
('autobahn', '==18.3.1')
('automat', '==0.6.0')
('backports.functools-lru-cache', '==1.5')
('baron', '==0.6.6')
('billiard', '==3.5.0.4')
('boto', '==2.47.0')
...

for x in parse_requirements('./requirements_git.txt', session=''):
    print((x.name, str(x.specifier)))
('ansiconv', '')
('django-qsstats-magic', '')
('dm.xmlsec.binding', '')
('jsonbfield', ''
...
```

ryanpetrello commented on awx/main/tests/functional/test_licenses.py:
```diff
@@ -0,0 +1,135 @@
+        # Check for GPL (forbidden) and LGPL (need to ship source)
+        # This is not meant to be an exhaustive check.
+        def check_license(file):
+            with open(file) as f:
```
This is a bit nitpick, but using `file` as a variable name is frowned upon, because it's a built-in.

wenottingham committed:
Add a test that we are including proper license files for all requirements.

wenottingham committed:
Clean up included licenses such that tests pass.

Rename ui licenses to '.txt' for consistency.
Update bundled code as appropriate.
Remove dead licenses and dev-only UI licenses.
Add additional python licenses from Azure & related updates.

wenottingham committed:
Add updates to UI license grabber from jlmitch5.
